### ISSUE TITLE
Installation Issues

### ISSUE THREAD

**Issue by makeclean** [raysect/source#255]
I followed the instructions on the documentation site - https://raysect.github.io/documentation/installation.html and then running the cornell_box demo failed, with import errors 

```Traceback (most recent call last):
  File "cornell_box.py", line 4, in <module>
    from raysect.primitive import Sphere, Box
  File "/home/adavis/.local/lib/python2.7/site-packages/raysect-0.5.4-py2.7.egg/raysect/primitive/__init__.py", line 30, in <module>
    from .box import Box```

I added the raysect egg folder into my PYTHONPATH. Box looks to be in primitive, so maybe this is out of date?

I expected to see some CI status from Travis/Circle but I didnt see any, I presume you do have CI but maybe its private? I figured it may hint what im doing wrong? I guess insideof there you run all the demos and all the nosetests, I guess it would be useful to expose that to users? I'll keep pluggin along in case its just an out of date file.

**Comment by mattngc** [comment]
Hi Andy,

Looks like you are using python 2.7? Raysect only works for Python 3.5+.

Matt

**Comment by makeclean** [comment]
Oh i see, I figured that one would need pip3, but the docs indicated pip, is that a convention thing?

**Comment by mattngc** [comment]
It depends on how your environment is set up. For me pip maps to python 3.5.

**Comment by makeclean** [comment]
Also, same issues im afraid 

```
Traceback (most recent call last):
  File "cornell_box.py", line 4, in <module>
    from raysect.primitive import Sphere, Box
  File "/home/adavis/.local/lib/python3.6/site-packages/raysect-0.5.4-py3.6.egg/raysect/primitive/__init__.py", line 30, in <module>
    from .box import Box
ModuleNotFoundError: No module named 'raysect.primitive.box'``

**Comment by mattngc** [comment]
The second problem was more to do with the documentation. The latest master doesn't automatically build with cython. Needs to be installed with `python setup.py install --use-cython`. The documentation needs to be updated to reflect this.

**Comment by makeclean** [comment]
Yeah, personally I think 'out of the box' it should just build, the users first impression is really critical here. I think there are a few issues

1) There was no indication from the build system that anything was wrong, that should obviously be rectified
2) I think its really a situation that shouldnt arise, it should use cython by default, an advanced option (i suspect for a more advanced user) should be to pull it from some precompiled location. I guess there are several problems with that too, is there guaranteed ABI compatibility for cython compiled on linux to osx?
3) As @mattngc says, the documentation should really reflect this.

**Comment by CnlPepper** [comment]
Raysect is a python extension and as such, strictly, only the .c files are necessary for the package to build. the cython stage just rebuilds the .c files. I'm a little stumped why it didn't compile! As it happens we've changed the default behaviour back to rebuilding the .c files in the next (development) release.

Annoyingly this does force the user to have cython installed, something I was trying to avoid.

**Comment by makeclean** [comment]
Could it be that I only had the Intel compiler suite available at the time? Could it expect GCC?

**Comment by CnlPepper** [comment]
Hmm, possibly. The c extension compliation is handled by a package in the standard library (setuptools). We don't define it's behaviour beyond providing it c files - see the setup.py file. It is possible it had an issue with the compiler.

**Comment by CnlPepper** [comment]
If you install it again at any point could you capture the log file? We can't recreate the error.

**Comment by mattngc** [comment]
Alex thinks this problem was caused by the c files not being included in the git repository. So the testing worked locally but failed when installed through pip or a clone. We have reverted to the original behaviour where cython compilation happens by default unless it is explicitly turned off.

Fixed in commit 6ef264b.
